### PR TITLE
Fix docker tag

### PR DESCRIPTION
**Pull request by oblitorum**
cc @niall-daley

### CODE REVIEW

File: build/bin/build-server [hunk 1]
```diff
@@ -43,9 +43,11 @@ build() {
     done
 
     if [ -z "${DOCKER_TAG}" ]; then
-        DOCKER_TAG=shatteredsilicon/ssm-server-fb:${ssm_version}
```
Comment (gordan-bobic) [inline build/bin/build-server:46]: Shouldn't the docker tag be set regardless?

Comment (gordan-bobic) [reply build/bin/build-server:46]: @oblitorum ^^

**Comment by oblitorum** [comment]
@gordan-bobic Don't forget this pr :)

File: build/bin/build-server (hunk 1 above)
Comment (oblitorum) [reply build/bin/build-server:46]: You mean the docker tags `shatteredsilicon/ssm-server:${ssm_version}` and `shatteredsilicon/ssm-server:latest` should always be set, even if people use a custom tag?

OK, I'll change it.

```diff
     if [ -z "${DOCKER_TAG}" ]; then
-        DOCKER_TAG=shatteredsilicon/ssm-server-fb:${ssm_version}
+        docker build --squash --no-cache -t shatteredsilicon/ssm-server:${ssm_version} -t shatteredsilicon/ssm-server:latest ${server_build_dir}/
+    else
+        docker build --squash --no-cache -t ${DOCKER_TAG} ${server_build_dir}/
     fi
-    docker build --squash --no-cache -t ${DOCKER_TAG} ${server_build_dir}/
+
```

**Comment by gordan-bobic** [comment]
No, I mean if docker_tag isn't set, you do docker build, but then there is a push happening on docker tag later on.
So if docker tag is not already set, should it not be set to something?

**Comment by oblitorum** [comment]
Oh, I see, you're right. But there are 2 tags `${ssm_version}` and `latest`, and `docker push` seems only support one tag or all tags, I have to split it into 2 `docker push`

**Comment by oblitorum** [comment]
Updated

```diff
     if [ -z "${DOCKER_TAG}" ]; then
-        DOCKER_TAG=shatteredsilicon/ssm-server-fb:${ssm_version}
-    fi
-    docker build --squash --no-cache -t ${DOCKER_TAG} ${server_build_dir}/
-
-    if [ -n "${PUSH_DOCKER}" ]; then
-        docker push ${DOCKER_TAG}
+        docker build --squash --no-cache -t shatteredsilicon/ssm-server:${ssm_version} -t shatteredsilicon/ssm-server:latest ${server_build_dir}/
+        if [ -n "${PUSH_DOCKER}" ]; then
+            docker push shatteredsilicon/ssm-server:${ssm_version}
+            docker push shatteredsilicon/ssm-server:latest
+        fi
+    else
+        docker build --squash --no-cache -t ${DOCKER_TAG} ${server_build_dir}/
+        if [ -n "${PUSH_DOCKER}" ]; then
+            docker push ${DOCKER_TAG}
+        fi
     fi
 }
```

**Comment by gordan-bobic** [comment]
Right, but here you are removing setting the docker tag:
-        DOCKER_TAG=shatteredsilicon/ssm-server-fb:${ssm_version}

And I don't see the docker tag being set anywhere else before lower down you are using the docker tag to push.

Is that right?

**Comment by oblitorum** [comment]
Oh, I moved that `docker push` part into the if/else block, if `DOCKER_TAG` is not set it will just `docker push` with constant values`${ssm_version}` and `latest`.

**Comment by gordan-bobic** [comment]
Oh, I see. Fair enough. Are you happy with the current version? All works as intended?

**Comment by oblitorum** [comment]
Yes, just tested it. And added some doc to README. Ready to be merged.

**Comment by niall-daley** [comment]
In general, the build script should not be pushing to docker hub at all. Remember that this script is public and will be used by others too. It would be best to simply remove the push parts of the script and leave that to the surrounding system. All this should produce is a docker image and a set of RPMs and SRPMs.

**Comment by oblitorum** [comment]
OK, got it, will remove that additional docker push feature

**Comment by gordan-bobic** [comment]
Agreed. Please move the pushing to dockerhub to a separate unrelated script.

We should treat this similarly to pushing packages to our yum repository. This shouldn't work anyway because the credentials won't be present, but either way, it should be a separate process so that somebody running the build doesn't get an error at the end.

**Comment by oblitorum** [comment]
There is a `PUSH_DOCKER` ENV variable, it will try to push the docker image only if this variable is set.

If we're going to remove it from build process, is it necessary to put it in a separate script? It's barely just a `docker push` command.

**Comment by oblitorum** [comment]
Oh, sorry, I guess you mean a unrelated script outside this project

**Pull request merged**